backend/inspirehep/migrator/tasks.py

```diff
@@ -291,14 +291,16 @@ def create_records_from_mirror_recids(self, recids):
             raise
         except Exception:
             LOGGER.exception("Cannot migrate record", recid=recid)
-            continue
+            raise
     try:
         db.session.commit()
     except Exception:
         LOGGER.exception(
             "Cannot commit migrated records", processed_records=list(processed_records)
         )
-    models_committed.connect(index_after_commit)
+        raise
+    finally:
```

Comment:
actually, there should be a `try`...`finally` around the whole block. If we `raise` from one of the previous exceptions, the signal won't be reattached. This doesn't matter currently as workers only process the migration and we index records separately, but will cause subtle bugs once we also do disambiguation in the workers.

Reply:
I think this should resolve the issue